File: tests/functional/Bots/Python/Consumers/CodeFirst/WaterfallHostBot/skills/teams_skill.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from enum import Enum
from botbuilder.schema import Activity
from skills.skill_definition import SkillDefinition


class TeamsSkill(SkillDefinition):
    class SkillAction(str, Enum):
        TEAMS_TASK_MODULE = "TeamsTaskModule"
        TEAMS_CARD_ACTION = "TeamsCardAction"
        TEAMS_CONVERSATION = "TeamsConversation"
        CARDS = "Cards"
        PROACTIVE = "Proactive"
        ATTACHMENT = "Attachment"
        AUTH = "Auth"
        SSO = "Sso"
        ECHO = "Echo"
        FILE_UPLOAD = "FileUpload"
        DELETE = "Delete"
        UPDATE = "Update"

    def get_actions(self):
        return self.SkillAction

    def create_begin_activity(self, action_id: str):
        if action_id not in self.SkillAction:
            raise Exception(f'Unable to create begin activity for "${action_id}".')

        # We don't support special parameters in these skills so a generic event with the
        # right name will do in this case.
        activity = Activity.create_event_activity()
        activity.name = action_id

        return activity

File: tests/functional/Bots/JavaScript/Consumers/CodeFirst/WaterfallHostBot/skills/teamsSkill.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

const { ActivityEx } = require('botbuilder-core');
const { SkillDefinition } = require('./skillDefinition');

const SkillAction = {
  TeamsTaskModule: 'TeamsTaskModule',
  TeamsCardAction: 'TeamsCardAction',
  TeamsConversation: 'TeamsConversation',
  Cards: 'Cards',
  Proactive: 'Proactive',
  Attachment: 'Attachment',
  Auth: 'Auth',
  Sso: 'Sso',
  Echo: 'Echo',
  FileUpload: 'FileUpload',
  Delete: 'Delete',
  Update: 'Update'
};

class TeamsSkill extends SkillDefinition {
  getActions () {
    return Object.values(SkillAction);
  }

  /**
   * @param {string} actionId
   */
  createBeginActivity (actionId) {
    if (!this.getActions().includes(actionId)) {
      throw new Error(`[TeamsSkill]: Unable to create begin activity for "${actionId}".`);
    }

    // We don't support special parameters in these skills so a generic event with the right name
    // will do in this case.
    const activity = ActivityEx.createEventActivity();
    activity.name = actionId;

    return activity;
  }
}

module.exports.TeamsSkill = TeamsSkill;
